src/System.Windows.Forms/src/System/Windows/Forms/CurrencyManager.cs

```diff
@@ -515,20 +517,20 @@ private void FindGoodRow()
     /// </summary>
     internal void SetSort(PropertyDescriptor property, ListSortDirection sortDirection)
     {
-        if (_list is IBindingList && ((IBindingList)_list).SupportsSorting)
+        if (_list is IBindingList bindingList && bindingList.SupportsSorting)
```

Comment:
We want to start using this syntax:
```suggestion
        if (_list is IBindingList { SupportsSorting: true } bindingList)
```
Ditto in a few places below